test/root/predicates/ERC20Predicate.test.js

```diff
@@ -0,0 +1,450 @@
+      const startExitTx = await utils.startExitWithBurntTokens(
+        contracts.ERC20Predicate,
+        { headerNumber, blockProof, blockNumber: block.number, blockTimestamp: block.timestamp, reference, logIndex: 1 }
+      )
+      const logs = logDecoder.decodeLogs(startExitTx.receipt.rawLogs)
+      // console.log(startExitTx, logs)
+      let log = logs[1]
+      log.event.should.equal('ExitStarted')
+      expect(log.args).to.include({
+        exitor: user,
+        token: childContracts.rootERC20.address,
+        burnt: true
+      })
+      utils.assertBigNumberEquality(log.args.amount, amount)
+    })
+  })
+
+  describe('startExit', async function() {
+    beforeEach(async function() {
+      contracts.withdrawManager = await deployer.deployWithdrawManager()
+      contracts.ERC20Predicate = await deployer.deployErc20Predicate()
+      const { rootERC20, childToken } = await deployer.deployChildErc20(accounts[0])
+      childContracts.rootERC20 = rootERC20
+      childContracts.childToken = childToken
+      // start = 0
+    })
+
+    it('reference: deposit - exitTx: fullBurn')
+
+    it('reference: incomingTransfer - exitTx: fullBurn', async function() {
+      await utils.deposit(
+        contracts.depositManager,
+        childContracts.childChain,
+        childContracts.rootERC20,
+        other,
+        amount
+      )
+      const { receipt } = await childContracts.childToken.transfer(user, amount, { from: other })
+      const { block, blockProof, headerNumber, reference } = await init(contracts.rootChain, receipt, accounts, start)
+      const { receipt: r } = await childContracts.childToken.withdraw(amount)
+      let exitTx = await web3Child.eth.getTransaction(r.transactionHash)
+      exitTx = await buildInFlight(exitTx)
+
+      const startExitTx = await utils.startExit(contracts.ERC20Predicate, headerNumber, blockProof, block.number, block.timestamp, reference, 1, /* logIndex */ exitTx)
+      const logs = logDecoder.decodeLogs(startExitTx.receipt.rawLogs)
+      // console.log(startExitTx, logs)
+      const log = logs[1]
+      log.event.should.equal('ExitStarted')
+      expect(log.args).to.include({
+        exitor: user,
+        token: childContracts.rootERC20.address
+      })
+      utils.assertBigNumberEquality(log.args.amount, amount)
+    })
+
+    it('reference: outgoingTransfer - exitTx: fullBurn', async function() {
+      await utils.deposit(
+        contracts.depositManager,
+        childContracts.childChain,
+        childContracts.rootERC20,
+        user,
+        amount
+      )
+
+      const { receipt } = await childContracts.childToken.transfer(other, halfAmount)
+      const { block, blockProof, headerNumber, reference } = await init(contracts.rootChain, receipt, accounts, start)
+
+      const { receipt: r } = await childContracts.childToken.withdraw(halfAmount)
+      let exitTx = await web3Child.eth.getTransaction(r.transactionHash)
+      exitTx = await buildInFlight(exitTx)
```

Comment:
we should replace these mock responses with actual data from the chain, else will be doomed by new ETH node versions and will be facing compatibility issues on the actual deployment. @atvanguard no?

Reply:
By _mock responses_, do you mean _responses local from parity/ganache_?
By _actual data on the chain_ do you mean, we should run these tests on ropsten?

Reply:
Right now we have these responses in JSON file, that should be live from local chain ganache/parity.

Reply:
I added these mock responses (JSON files) when I wrote some unit tests for libraries. These aren't being used right now and tests are in fact running on local ganache/parity.